lib/dal/src/schema/variant.rs

```diff
@@ -1051,20 +1059,13 @@ impl SchemaVariant {
     pub async fn upsert_leaf_function(
         ctx: &DalContext,
         schema_variant_id: SchemaVariantId,
-        component_id: Option<ComponentId>,
         leaf_kind: LeafKind,
         input_locations: &[LeafInputLocation],
         func: &Func,
     ) -> SchemaVariantResult<AttributePrototypeId> {
         let leaf_item_prop_id =
             SchemaVariant::find_leaf_item_prop(ctx, schema_variant_id, leaf_kind).await?;
 
-        if component_id.is_some() {
-            // NOTE(nick): replaced a "unimplemented" here with an error, but we will need to think
-            // about whether we want this in the future.
-            return Err(SchemaVariantError::ComponentContextNotSupportedForLeafFunctions);
-        }
```

Comment:
Removed all unused component-specific logic related to leaf upsertion other than automatic changes when leaf functions are added